client/app/visualizations/counter/counterTypes.js

```diff
@@ -0,0 +1,51 @@
+import { map, max, min, sumBy } from "lodash";
+
+// 0 - special case, use first record
+// 1..N - 1-based record number from beginning (wraps if greater than dataset size)
+// -1..-N - 1-based record number from end (wraps if greater than dataset size)
+function getRowNumber(rowNumber, rowsCount) {
+  rowNumber = parseInt(rowNumber, 10) || 0;
+  if (rowNumber === 0) {
+    return rowNumber;
+  }
+  const wrappedIndex = (Math.abs(rowNumber) - 1) % rowsCount;
+  return rowNumber > 0 ? wrappedIndex : rowsCount - wrappedIndex - 1;
+}
+
+export default {
+  unused: {
+    name: "Unused",
+    getValue: () => [undefined, null],
```

Comment:
I changed `returnsColumnNames` option name to `canReturnRow` - unfortunately I have no better idea for it :slightly_frowning_face: Also, added a comment that explains options for counter types.

Reply:
IMO ideally it should have `columnName` in the name to seem related to that setting, but I also don't have a better idea. Let's go with that and if eventually we add more conditional format options, we review that. Thanks for the explanation comment! :slightly_smiling_face:

Reply:
Anything with `columnName` looks not right because `getValue` actually returns not just a column names, but entire row (associated with this counter value).